packages/block-library/src/navigation/edit/index.js

```diff
@@ -490,11 +549,18 @@ function Navigation( {
 					isResolvingCanUserCreateNavigationMenu={
 						isResolvingCanUserCreateNavigationMenu
 					}
-					onFinish={ ( post ) => {
-						if ( post ) {
-							setRef( post.id );
+					onFinish={ ( navPostOrClassicMenu ) => {
+						const isClassicMenu = navPostOrClassicMenu.hasOwnProperty(
```

Comment:
Could that be a separate function? My reasoning:

* this logic is repeated in two places
* I'd rather have future us find that function when we need it, than missing this very specific part of edit/index.js and invent another way of checking if something is a classic menu

Reply:
Yes I agree. Looks like overtime the signatures of this handler become identical. That's good news as the incremental refactoring is obviously making the codebase easier to reason about.

I've extracted a common utility.